### DISCUSSION TITLE
Loading private npm packages only works for the 0th item in the `repositories` array

### DISCUSSION THREAD

**Discussion by richardkazuomiller** [renovatebot/renovate#15293] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.22.2

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

My company has some internal packages we update with Renovate and it stopped working for everything the 0th item in the repositories array. It worked until upgrading from `31.50.2` to `31.97.3`.

I noticed that when I ran `renovate my-company/repo-name` it always worked. I then tried changing `repositories` in my configuration to just a single item, and that also worked. I then added one more repo so the array was length 2, and the 0th item succeeded while the other one failed. I also tried switching the order, and it's always the 0th repo that works.

The debug logs look like the request doesn't include any authentication headers.

Creating a reproduction repo is difficult, since it involves private packages.

Upgrading to the latest v32 didn't fix the issue. 

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Dependency lookup failure: unauthorized (repository={my-company}/{my-repo})
       "packageUrl": "https://npm.pkg.github.com/{my-company}/@{my-company}%2F{internal-dependency}",
       "err": {
         "name": "HTTPError",
         "code": "ERR_NON_2XX_3XX_RESPONSE",
         "timings": {
           "start": 1650041157154,
           "socket": 1650041157154,
           "lookup": 1650041157154,
           "connect": 1650041157165,
           "secureConnect": 1650041157182,
           "upload": 1650041157182,
           "response": 1650041157361,
           "end": 1650041157361,
           "abort": 1650041157362,
           "phases": {
             "wait": 0,
             "dns": 0,
             "tcp": 11,
             "tls": 17,
             "request": 0,
             "firstByte": 179,
             "download": 0,
             "total": 208
           }
         },
         "message": "Response code 401 (Unauthorized)",
         "stack": "HTTPError: Response code 401 (Unauthorized)\n    at Request.<anonymous> (/Users/richardmiller/srv/webtech-renovate/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/32.22.2 (https://github.com/renovatebot/renovate)",
             "accept": "application/json",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://npm.pkg.github.com/{my-company}/@{my-company}%2F{internal-dependency}",
           "hostType": "npm",
           "username": "",
           "password": "",
           "method": "GET",
           "http2": false
         },
         "response": {
           "statusCode": 401,
           "statusMessage": "Unauthorized",
           "body": "",
           "headers": {
             "access-control-allow-methods": "GET, HEAD, OPTIONS",
             "access-control-allow-origin": "*",
             "content-security-policy": "default-src 'none';",
             "server": "GitHub Registry",
             "strict-transport-security": "max-age=31536000;",
             "www-authenticate": "Basic realm=\"GitHub Package Registry\"",
             "x-content-type-options": "nosniff",
             "x-frame-options": "DENY",
             "x-xss-protection": "1; mode=block",
             "date": "Fri, 15 Apr 2022 16:45:57 GMT",
             "content-length": "0",
             "x-github-request-id": "DF08:5D03:2DFF2:51E5C:6259A145",
             "connection": "close"
           },
           "httpVersion": "1.1"
         }
       },
       "statusCode": 401,
       "packageName": "@{my-company}/{internal-dependency}"
DEBUG: Failed to look up dependency @{my-company}/{internal-dependency} (repository={my-company}/{my-repo}, packageFile=package.json, dependency=@{my-company}/{internal-dependency})
DEBUG: Dependency lookup failure: unauthorized (repository={my-company}/{my-repo})
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by richardkazuomiller** [comment]
I did a binary-ish search and it looks like `31.92.0` is the last version where this worked for me and the issue was introduced in `31.92.1` <https://github.com/renovatebot/renovate/compare/31.92.0...31.92.1>

**Comment by rarkins** [comment]
We need to know more about how your are configuring Renovate

**Comment by richardkazuomiller** [comment]
My config looks like this:

```javascript
module.exports = {
  dependencyDashboard: true,
  platform: 'github',
  enabledManagers: [
    'npm'
  ],
  token: process.env.GITHUB_TOKEN,
  npmrc: `@{org}:registry=https://npm.pkg.github.com/{org}\n//npm.pkg.github.com/:_authToken=${process.env.GITHUB_TOKEN}`,
  prConcurrentLimit: 5,
  branchConcurrentLimit: 16,
  onboardingPrTitle: 'chore(deps): configure renovate',
  onboardingCommitMessage: 'chore(deps): configure renovate',
  labels: [
    'renovate'
  ],
  updateLockFiles: true,
  extends: [
    'github>whitesource/merge-confidence:beta'
  ],
  packageRules: [
    ...rules
  ],
  repositories: [
    // 15 or so private repos
  ]
};
```

It seems like using npmrc might be the relevant part to what broke in that version.

Edit: added npmrc

**Comment by viceice** [comment]
how does your npmrc config option look like?

**Comment by richardkazuomiller** [comment]
The `npmrc` in my `config.js` is like this, with `{org}` being the actual name of the org.

```
npmrc: `@{org}:registry=https://npm.pkg.github.com/{org}\n//npm.pkg.github.com/:_authToken=${process.env.GITHUB_TOKEN}`,
```

**Comment by rarkins** [comment]
So your `npmrc` value within `config.js` is being applied to the first repo, and none other?

**Comment by richardkazuomiller** [comment]
That's what it looks like, yes.

**Comment by rarkins** [comment]
For the repos you tested with, do they have any `npmrc` configuration within `renovate.json` or `.npmrc` file within the repo itself?

**Comment by richardkazuomiller** [comment]
There is, in each repo, yes. Each has a one line `.npmrc` file with `@{org}:registry=...`

**Comment by rarkins** [comment]
Do you observe any difference if [`npmrcMerge`](https://docs.renovatebot.com/configuration-options/#npmrcmerge) is set to `true`?

**Comment by richardkazuomiller** [comment, marked as answer]
It appears to be working as expected if I set `npmMerge` to `true`! 😄
